### ISSUE TITLE
#pragma causes more warnings than it solves (Trac #1827)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1819]
r149256/IceCube was supposed to suppress warnings but clang on osx dosn't understand the #pragma and causes a lot of these warnings:

```text
In file included from /Users/kmeagher/icecube/combo/src/bayesian-priors/private/bayesian-priors/I3ZenithWeight.cxx:16:
    In file included from /Users/kmeagher/icecube/combo/src/bayesian-priors/public/bayesian-priors/I3ZenithWeight.h:23:
    In file included from /Users/kmeagher/icecube/combo/src/dataclasses/public/dataclasses/physics/I3Particle.h:16:
    /Users/kmeagher/icecube/combo/src/dataclasses/public/dataclasses/physics/I3ParticleID.h:50:32:
    warning: unknown warning group '-Wmaybe-uninitialized'
  , ignored [-Wunknown-pragmas]
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
                               ^
/Users/kmeagher/icecube/combo/src/dataclasses/public/dataclasses/physics/I3ParticleID.h:55:32:
    warning: unknown warning group '-Wmaybe-uninitialized'
  , ignored [-Wunknown-pragmas]
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
                               ^
/Users/kmeagher/icecube/combo/src/dataclasses/public/dataclasses/physics/I3ParticleID.h:60:32:
    warning: unknown warning group '-Wmaybe-uninitialized'
  , ignored [-Wunknown-pragmas]
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
                               ^
3 warnings generated.
```
 

Migrated from https://code.icecube.wisc.edu/ticket/1827
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:12:38", 
    "description": "r149256/IceCube was supposed to suppress warnings but clang on osx dosn't understand the #pragma and causes a lot of these warnings:\n{{{\nIn file included from /Users/kmeagher/icecube/combo/src/bayesian-priors/private/bayesian-priors/I3ZenithWeight.cxx:16:\n    In file included from /Users/kmeagher/icecube/combo/src/bayesian-priors/public/bayesian-priors/I3ZenithWeight.h:23:\n    In file included from /Users/kmeagher/icecube/combo/src/dataclasses/public/dataclasses/physics/I3Particle.h:16:\n    /Users/kmeagher/icecube/combo/src/dataclasses/public/dataclasses/physics/I3ParticleID.h:50:32:\n    warning: unknown warning group '-Wmaybe-uninitialized'\n  , ignored [-Wunknown-pragmas]\n#pragma GCC diagnostic ignored \"-Wmaybe-uninitialized\"\n                               ^\n/Users/kmeagher/icecube/combo/src/dataclasses/public/dataclasses/physics/I3ParticleID.h:55:32:\n    warning: unknown warning group '-Wmaybe-uninitialized'\n  , ignored [-Wunknown-pragmas]\n#pragma GCC diagnostic ignored \"-Wmaybe-uninitialized\"\n                               ^\n/Users/kmeagher/icecube/combo/src/dataclasses/public/dataclasses/physics/I3ParticleID.h:60:32:\n    warning: unknown warning group '-Wmaybe-uninitialized'\n  , ignored [-Wunknown-pragmas]\n#pragma GCC diagnostic ignored \"-Wmaybe-uninitialized\"\n                               ^\n3 warnings generated.\n}}}\n ", 
    "reporter": "kjmeagher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067158057333", 
    "component": "combo core", 
    "summary": "#pragma causes more warnings than it solves", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-08-18T09:18:33", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "cleanup"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160818T13:53:12: nega commented:

> In this case, the easy answer is to bump up to a more general warning group (probably `-Wall`, here). Each section of code should also end with a `#pragma GCC diagnostic pop` (a `pop` without a corresponding `push` will return to the "command line state").
> 
> `pop`/`push` is GCC version dependent, but as long as it's available after 4.8.1 (the new minimum) we're good. On prior versions it will just be an "unknown pragma" warning and we don't gain or lose anything in that case.

**Comment by icecube-trac** [comment]
Trac update at 20160818T14:20:55: 
* olivas changed status from "new" to "accepted"
* olivas changed milestone from "" to "August 2016"
* olivas changed type from "defect" to "cleanup"
* olivas changed component from "cmake" to "icecube offline"

**Comment by icecube-trac** [comment]
Trac update at 20160818T15:15:11: 
* olivas commented:

> In changeset 149349:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="149349"
> only suppress for gcc and restore the command line options at the end. closes #1819.
> ```


* olivas changed status from "accepted" to "closed"
* olivas changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:12:38: 
* olivas changed milestone from "August 2016" to ""
* olivas commented:

> Milestone August 2016 deleted
